nomad/structs/services.go

```diff
@@ -1353,8 +1353,10 @@ func (p *ConsulGatewayProxy) Copy() *ConsulGatewayProxy {
 	}
 
 	bindAddresses := make(map[string]*ConsulGatewayBindAddress, len(p.EnvoyGatewayBindAddresses))
-	for k, v := range p.EnvoyGatewayBindAddresses {
-		bindAddresses[k] = v.Copy()
+	if p.EnvoyGatewayBindAddresses != nil {
```

Comment:
this shouldn't be necessary, `for` will skip over an empty map already
https://play.golang.org/p/KZGGfy4XTiQ